core/domain/beam_job_services_test.py

```diff
@@ -182,7 +182,7 @@ def test_get_beam_job_runs_with_refresh(self):
         self.assert_domains_equal_models(
             beam_job_runs, updated_beam_job_run_models)
 
-    def test_get_beam_job_runs_with_refresh_of_non_dataflow_job(self):
+    def test_get_beam_job_runs_with_refresh_of_synchronous_job(self):
```

Comment:
Please add a comment explaining what a "synchronous job" is. 

I'm still not entirely certain about the terminology -- the job is the same job in both dev and prod, and synchronicity refers to the environment rather than the job itself. But I think an explanatory comment here and below would be enough to clear any misunderstandings.

Reply:
How's this?

Reply:
It helps, but what I mean is explaining what a synchronous job is, or where the reader can find the definition of one.

There's just basically not enough context for someone who's not on this team to understand what is going on, IMO.

Reply:
OK, I've decided to try and resolve this by making it an official attribute of `BeamJobRun` with its own documentation. Would that work?

Reply:
Yup, just call it "synchronous" instead and be consistent about how you're describing the context for the sync and async cases.

Reply:
Opted for `job_is_synchronous` for consistency. Done with documentation as well.